src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,225 @@
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::shape::text::glyph::system::Line;
+use ensogl::display::shape::text::glyph::system::GlyphSystem;
+use ensogl::display::Sprite;
+use ensogl::gui::component;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const TEXT_SIZE         : f32 = 12.0;
+const GLYPH_WIDTH       : f32 = 7.224_609_4; // FIXME[dg] hardcoded literal. Copied from port.rs
+const VERTICAL_MARGIN   : f32 = GLYPH_WIDTH * 2.0;
+const HORIZONTAL_MARGIN : f32 = GLYPH_WIDTH;
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+mod background {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style, selection:f32) {
+            let bg_color = color::Rgba::new(0.0,0.0,0.0,0.000_01);
+            Plane().fill(bg_color).into()
+        }
+    }
+}
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef,Copy)]
+#[allow(missing_docs)]
+pub struct FrpInputs {}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
+    pub fn new(_network:&frp::Network) -> Self {
+        Self{}
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    pub selected : frp::Source
+}
+
+impl FrpOutputs {
+    /// Create new FrpOutputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend!{ network
+            def selected = source();
+        }
+        Self{selected}
+    }
+}
+
+
+
+// ===========
+// === Frp ===
+// ===========
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct Frp {
+    pub inputs  : FrpInputs,
+    pub outputs : FrpOutputs,
+    pub network : frp::Network,
+}
+
+impl Deref for Frp {
+    type Target = FrpInputs;
+    fn deref(&self) -> &Self::Target {
+        &self.inputs
+    }
+}
+
+impl Default for Frp {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Frp {
+    /// Create new Frp.
+    pub fn new() -> Self {
+        let network = frp::Network::new();
+        let inputs  = FrpInputs::new(&network);
+        let outputs = FrpOutputs::new(&network);
+        Self{network,inputs,outputs}
+    }
+}
+
+
+
+// ========================
+// === BreadcrumbModel ===
+// ========================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct BreadcrumbModel {
+    logger         : Logger,
+    display_object : display::object::Instance,
+    view           : component::ShapeView<background::Shape>,
+    glyph_system   : GlyphSystem,
+    label          : Line,
+    pub name       : Rc<String>
```

Comment:
Not sure how this fits into the big picture (with integration and so on) but someone will need to know more about crumb than just its displayed string.

(e.g. the expression id for the local call)